generator/src/main/java/com/scottlogic/deg/generator/inputs/MainConstraintReader.java

```diff
@@ -27,11 +28,13 @@ public Constraint apply(
             throw new InvalidProfileException("Constraint is null");
         }
 
-        if (dto.is != null) {
-            ConstraintReader subReader = this.atomicConstraintReaderLookup.getByTypeCode(dto.is);
+        String isConstraintValue = readIsConstraint(dto);
+
+        if (isConstraintValue != null) {
+            ConstraintReader subReader = this.atomicConstraintReaderLookup.getByTypeCode(isConstraintValue);
 
             if (subReader == null) {
-                throw new InvalidProfileException("Couldn't recognise constraint type from DTO: " + dto.is);
+                throw new InvalidProfileException("Couldn't recognise constraint type from DTO: " + isConstraintValue);
```

Comment:
You could use `getIs()` here and remove `readIsConstraint` if there is only one exception type.